### ISSUE TITLE
您好，我想请教一下关于corss-document-attention的几个问题

### ISSUE THREAD

**Issue by tyistyler** [SVAIGBA/CDKGen#1]
大佬您好，我想请教您一下有关corss-document-attention的几个问题：
请问在word_memory中，
u = torch.mean(self.word_embedding_b(word_seq),0)
p = F.softmax(torch.matmul(u, self.word_embedding_a), dim=1)
中的u和p是怎么变化的呢，我在您的代码中没有发现其中的奥妙
以及word_embedding_a、word_embedding_b、word_embedding_c是怎么变化的？

**Comment by shizhediao** [comment]
Hi, thanks for your question.
Because word_embedding_a  and word_embedding_c are nn.Parameter, they will be updated during training.
As for word_embedding_b, it is initialized randomly using the OpenNMT default embedding.

**Comment by tyistyler** [comment]
thanks, is the  word_embedding_b is a trainable matrix?

**Comment by shizhediao** [comment]
yes, it is.
